### ISSUE TITLE
Any plans or timelines to release gshard code?

### ISSUE THREAD

**Issue by ngoyal2707** [tensorflow/lingvo#229]
Great work on gshard, curious if there are any plans to open-source it and if so, timeline?

**Comment by DanielTizon** [comment]
Yes, We are thinking about to use GPT-3 or GShard, and open source code and models is really important for us.

**Comment by orsharir** [comment]
Seems like they did: https://github.com/tensorflow/lingvo/commit/45da620d670583b07b62108c6d81e31e05364cc4

**Comment by bignamehyp** [comment]
Our code has been open sourced. The instructions to run gshard dense transformer on gcp tpus are described here:
https://github.com/tensorflow/lingvo/tree/master/lingvo/tasks/lm

**Comment by gbxu** [comment]
> Our code has been open sourced. The instructions to run gshard dense transformer on gcp tpus are described here:
> https://github.com/tensorflow/lingvo/tree/master/lingvo/tasks/lm

No GPU-supported implementation?
